lib/frame-manager.js

```diff
@@ -32,7 +31,7 @@ function FrameManager(window) {
   this.on('removeListener', unsubscribe);
 
   function subscribe(eventName) {
-    if (!subscribed && eventName === 'data') {
+    if (!self.listenerCount('data') && eventName === 'data') {
```

Comment:
Oh, geez, I really lost track of cleaning up that `subscribed` flag! This approach is way better 👍